internal/transport/http2_server.go

```diff
@@ -389,6 +390,41 @@ func (t *http2Server) operateHeaders(frame *http2.MetaHeadersFrame, handle func(
 			if timeout, err = decodeTimeout(hf.Value); err != nil {
 				headerError = true
 			}
+		// "Transports must consider requests containing the Connection header
+		// as malformed." - A41
+		case "connection":
+			t.controlBuf.put(&cleanupStream{
```

Comment:
Log error here and the other reset streams.

Reply:
Me and Doug discussed this in meeting, he mentioned that he wants to send back a GRPC Status (or clause in A41) in the case where this server knows the client is talking grpc, which you can write a message to. That is orthogonal to what you are asking here for logs right? Also any thoughts on this?

Reply:
What I said is to `logger.Infof`.

You only know if the request is gRPC after processing all the fields. So for that, you will need to do all these checks after the loop.
Just append all authorities and hosts to mdata, and check the length at the end. `earlyAbortStream` would send the status I believe.

Also, in that case, all non-gRPC requests will be reset with `http2.ErrCodeProtocol`. We don't care about the other headers.

Reply:
Why are you bringing up "other headers"? Yup, exactly what he said :). Will add.

Reply:
Switched to the logic discussed in chat.